index.js

```diff
@@ -19,23 +20,35 @@ const InterestingVariables = [
 ]
 
 function findVcvarsall() {
+    // use vswhere
+    let path = child_process.execSync('vswhere -products * -latest -prerelease -find **/Auxiliary/Build/vcvarsall.bat').toString().trim()
+    if (path && fs.existsSync(path)) {
+        return path
+    }
+
     const programFiles = process.env['ProgramFiles(x86)']
     // Given the order of each list it should check
     // for the more recent versions first and the
     // highest grade edition first.
     for (const ver of VERSIONS) {
         for (const ed of EDITIONS) {
-            const path = `${programFiles}\\Microsoft Visual Studio\\${ver}\\${ed}\\VC\\Auxiliary\\Build\\vcvarsall.bat`
+            path = `${programFiles}\\Microsoft Visual Studio\\${ver}\\${ed}\\VC\\Auxiliary\\Build\\vcvarsall.bat`
             if (fs.existsSync(path)) {
                 return path
             }
         }
     }
     // Special case for Visual Studio 2015 (and maybe earlier), try it out too.
-    const path = `${programFiles}\\Microsoft Visual C++ Build Tools\\vcbuildtools.bat`
+    path = `${programFiles}\\Microsoft Visual C++ Build Tools\\vcbuildtools.bat`
     if (fs.existsSync(path)) {
         return path
     }
+    // use vswhere
+    path = child_process.execSync('vswhere -products * -latest -prerelease -find **/vcbuildtools.bat').toString().trim()
+    if (path && fs.existsSync(path)) {
+        return path
+    }
```

Comment:
Shouldn't this be performed before the probing check, like for `vsvarsall.bat`?

Reply:
Fixed in https://github.com/ilammy/msvc-dev-cmd/pull/11/commits/d0aaa5cdcecb382db69c324f57debb56b71d3929